### DISCUSSION TITLE
Any modular examples that contain login logic

### DISCUSSION THREAD

**Discussion by timelery** [pointfreeco/swift-composable-architecture#2449] [Q&A]
Is anyone aware of any recent TCA examples that contain login logic?

**Comment by kamcma** [comment]
[The tic-tac-toe example](https://github.com/pointfreeco/swift-composable-architecture/tree/f0bae57f361b3611b48d7db0b78393e5c78be1fc/Examples/TicTacToe) contains a faux login. (It doesn't actually log in to anything, but it can give you an idea of the shape such functionality might take.)

**Reply by timelery** [reply]
This example does not contain an appDelegate. The way Store using an enum for State is not common in any reasonably sized app. As such it is not a good example of a common app architecture requiring authentication. Is there anything elese out there?

**Reply by kamcma** [reply]
> The way Store using an enum for State is not common in any reasonably sized app.

This is quite valid to do in TCA. [TCA ships tools specifically to deal with enum state](https://pointfreeco.github.io/swift-composable-architecture/1.2.0/documentation/composablearchitecture/reducer/ifcaselet(_:action:then:fileid:line:)#discussion), and [the navigation tools make heavy use of putting big chunks of state in enums](https://pointfreeco.github.io/swift-composable-architecture/1.2.0/documentation/composablearchitecture/treebasednavigation#Enum-state). What is your concern?

>This example does not contain an appDelegate.

Not sure what this has to do with it.

**Comment by mbrandonw** [comment]
Hi @timelery, TCA has no specific opinion about "login logic" in particular, and so it will all depend on what your requirements are. Typically whatever you currently do for login in your app can be adapted to work with TCA too.

Perhaps you can explain in detail what your requirements are for login logic and how you approach this in your applications today. That might make it easier to describe how such a system can be used in TCA.
